logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -396,6 +415,15 @@ private void updateOldestSegmentReference() throws IOException {
             oldestSegmentTimestamp = Optional.empty();
             return;
         }
+
+        boolean previousPathEqualsToCurrent = previousOldestSegmentPath != null && // optional is present
```

Comment:
Instead of checking null, would it be better to initialize with Optional.empty() ?
```java
private Optional<Path> oldestSegmentPath = Optional.empty();
```